## Infomation Visualization Project 2 Data

### This page is for the convenience of code reuse and for saving useful information. You might find it helpful for understanding provided data files by refering to cells below.

In [17]:
import csv
filename = "./NCWIT_DataV2_RawData.csv"
with open(filename) as f:
    reader = csv.reader(f)
    data_list = list(reader)
    cnt1 = 0
    cnt2 = 0
    for row in data_list[1:2908]: 
        # 2907 valid records in total
        # line 2907 only has a record on a single column
        # extension service
        if "Ex" in row[4]: cnt1 += 1
        if "All" not in row[4]: cnt2 += 1
    # All institutions with extension services from NCWIT are academic alliance
    print("Extension Services: ", cnt1)
    print("Academic Alliance: ", 2907-cnt2)
        

Extension Services:  1173
Academic Alliance:  2905


## 1. Retention and NCWIT services

In [31]:
# sophomores females enrolled in different(68) and same(69) major
print(data_list[0][68])
print(data_list[0][69])

Sophomores, Female: Enrolled in DIFFERENT MAJOR (Sph. F)
Sophomores, Female: Enrolled in SAME MAJOR (Sph. F)


In [60]:
cnt_3 = 0
for row in data_list[1:2908]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major 
    if row[68] != '' and row[69] != '': cnt_3 += 1
print(cnt_3)

557


In [55]:
csvFile2 = open('Sophmore_major_changed.csv','w', newline='')
writer = csv.writer(csvFile2)
writer.writerow(["NCWIT Participant","Institution","Different", "Same"])
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major 
    if row[68] != '' and row[69] != '': 
        writer.writerow([row[4],row[5],row[68],row[69]])
csvFile2.close()

In [44]:
# Freshmen females enrolled in different(68) and same(69) major
print(data_list[0][40])
print(data_list[0][41])

Freshmen, Female: Enrolled in DIFFERENT MAJOR (Fshm F)
Freshmen, Female: Enrolled in SAME MAJOR (Fshm F)


In [61]:
cnt_4 = 0
for row in data_list[1:2908]:
    # number of valid records for Freshman, Female: Enrolled in DIFFERENT/SAME Major 
    if row[40] != '' and row[41] != '': cnt_4 += 1
print(cnt_4)

510


In [56]:
csvFile3 = open('Freshman_major_changed.csv','w', newline='')
writer = csv.writer(csvFile3)
writer.writerow(["NCWIT Participant","Institution","Different", "Same"])
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major 
    if row[40] != '' and row[41] != '': 
        writer.writerow([row[4],row[5],row[40],row[41]])
csvFile3.close()

The **different majors** might be related to Information Technology as well, instead of Art, Law, Economics or something else. In this case we will get less benefits from these two columns. 

In [51]:
# Freshmen females VS sophomores females total declared majors
print(data_list[0][48])
print(data_list[0][76])

Freshmen, Female: Total Declared Majors (Fshm F)
Sophomores, Female: Total Declared Majors (Sph. F)


In [62]:
cnt_5 = 0
for row in data_list[1:2908]:
    # number of valid records for Freshman, Female: Enrolled in DIFFERENT/SAME Major 
    if row[48] != '' and row[76] != '': cnt_5 += 1
print(cnt_5)

721


In [81]:
csvFile4 = open('Freshman_Sophomore_total.csv','w', newline='')
writer = csv.writer(csvFile4)
writer.writerow(["NCWIT Participant","Institution","School Year","Freshman", "Sophomore"])
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major 
    if row[48] != '' and row[76] != '': 
        writer.writerow([row[4],row[5],row[7],row[48],row[76]])
csvFile4.close()

For calculating retention rate.

## 2. Male/Female GPA with / without Extension Services

In [79]:
# females VS males GPA
print(data_list[0][12])
print(data_list[0][26])
print(data_list[0][39])
print(data_list[0][53])
print(data_list[0][67])
print(data_list[0][81])
print(data_list[0][95])
print(data_list[0][109])
print(data_list[0][123])
print(data_list[0][137])
print(data_list[0][179])
print(data_list[0][193])

Enroll, Female: Average HS GPA (Enrl F)
Enroll, Male: Average HS GPA (Enrl M)
Freshmen, Female: Cumulative GPA (Fshm F)
Freshmen, Male: Cumulative GPA (Fshm M)
Sophomores, Female: Cumulative GPA (Sph. F)
Sophomores, Male: Cumulative GPA (Sph. M)
Juniors, Female: Cumulative GPA (Jun. F)
Juniors, Male: Cumulative GPA (Jun. M)
Seniors, Female: Cumulative GPA (Sen. F)
Seniors, Male: Cumulative GPA (Sen. M)
Totals, Female: Cumulative GPA (Tot. F)
Totals, Male: Cumulative GPA (Tot. M)


In [83]:
csvFile5 = open('Male_Female_GPA_raw.csv','w', newline='')
writer = csv.writer(csvFile5)
writer.writerow([data_list[0][4], data_list[0][5],data_list[0][7], data_list[0][12],data_list[0][26],data_list[0][39], data_list[0][53],data_list[0][67], data_list[0][81],data_list[0][95], data_list[0][109],data_list[0][123], data_list[0][137], data_list[0][179], data_list[0][193]])
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major 
    writer.writerow([row[4],row[5],row[7],row[12],row[26],row[39],row[53],row[67],row[81],row[95],row[109],row[123],row[137],row[179],row[193]])
csvFile5.close()

This file includes data about GPA (high school, grade 1, 2, 3, 4, total) with **missing fileds**.

In [85]:
csvFile6 = open('Male_Female_GPA_total.csv','w', newline='')
writer = csv.writer(csvFile6)
writer.writerow([data_list[0][4], data_list[0][5],data_list[0][7], data_list[0][179], data_list[0][193]])
cnt6 = 0
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major 
    if row[179] != '' and row[193] != '':
        writer.writerow([row[4],row[5],row[7],row[179],row[193]])
        cnt6 += 1
csvFile6.close()
print(cnt6)

812


## 3. Relationship between SAT/ACT/HSGPA and GPA

Maybe we only need to compare the SAT/ACT/HSGPA with Freshmen Cumulative GPA since there will be a lot of work do calculation on the data over time.

In [89]:
print(data_list[0][9])
print(data_list[0][23])
print(data_list[0][13])
print(data_list[0][27])
print(data_list[0][12])
print(data_list[0][26])
print(data_list[0][39])
print(data_list[0][53])

Enroll, Female: Avg. ACT Math Score (Enrl F)
Enroll, Male: Avg. ACT Math Score (Enrl M)
Enroll, Female: Avg. SAT Math Score (Enrl F)
Enroll, Male: Avg. SAT Math Score (Enrl M)
Enroll, Female: Average HS GPA (Enrl F)
Enroll, Male: Average HS GPA (Enrl M)
Freshmen, Female: Cumulative GPA (Fshm F)
Freshmen, Male: Cumulative GPA (Fshm M)


In [96]:
csvFile7 = open('ACT_SAT_HS_FreshmenGPA_raw.csv','w', newline='')
writer = csv.writer(csvFile7)
writer.writerow([data_list[0][9], data_list[0][23],data_list[0][13], data_list[0][27], data_list[0][12], data_list[0][26], data_list[0][39], data_list[0][53]])
cnt7 = 0
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major
    if not (row[23] == '' and row[9] == '' and row[13] == '' and row[27] == '' and row[12] == '' and row[26] == '' and row[39] == '' and row[53] == ''):
        writer.writerow([row[9],row[23],row[13],row[27],row[12],row[26],row[39],row[53]])
        cnt7 += 1
csvFile7.close()
print(cnt7)

815


In [98]:
csvFile8 = open('ACT_SAT_HS_FreshmenGPA.csv','w', newline='')
writer = csv.writer(csvFile8)
writer.writerow([data_list[0][9], data_list[0][23],data_list[0][13], data_list[0][27], data_list[0][12], data_list[0][26], data_list[0][39], data_list[0][53]])
cnt8 = 0
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major
    if (row[23] != '' and row[9] != '' and row[13] != '' and row[27] != '' and row[12] != '' and row[26] != '' and row[39] != '' and row[53] != ''):
        writer.writerow([row[9],row[23],row[13],row[27],row[12],row[26],row[39],row[53]])
        cnt8 += 1
csvFile8.close()
print(cnt8)

154


If we discard all rows with missing fields, we only get 154 valid records, which sounds enough for visualization...Maybe not...

## 4. Number of Enrollments Changed Over Years

In [100]:
print(data_list[0][7])
print(data_list[0][19])
print(data_list[0][20])
print(data_list[0][33])
print(data_list[0][34])


School Year
Enroll, Female: New Enrollments (Enrl F)
Enroll, Female: Transfer Students (Enrl F)
Enroll, Male: New Enrollments (Enrl M)
Enroll, Male: Transfer Students (Enrl M)


In [102]:
csvFile9 = open('Enrollment_Number_years.csv','w', newline='')
writer = csv.writer(csvFile9)
writer.writerow([data_list[0][7], data_list[0][19],data_list[0][20], data_list[0][33], data_list[0][34]])
cnt9 = 0
for row in data_list[1:2907]:
    # number of valid records for Sophomores, Female: Enrolled in DIFFERENT/SAME Major
    if row[19] != '' and row[33] != '':
        writer.writerow([row[7],row[19],row[20],row[33],row[34]])
        cnt9 += 1
csvFile9.close()
print(cnt9)

887


The Numbers of transfer students might be helpful. But it could be missing in this files.